src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -2293,6 +2294,36 @@ public void testCaseMultipleWhenElse() throws DatabaseEngineException {
         assertEquals("COL5 must be KEK", "KEK", result.get(4).get("case").toString());
     }
 
+    @Test
+    public void testConcat() throws DatabaseEngineException {
+        test5Columns();
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "xpto")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 3).set("COL5", "xpto")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 4).set("COL5", "teste")
+                .build());
+        engine.persist("TEST", entry().set("COL1", 5).set("COL5", "pomme de terre")
+                .build());
+
+        final Query query =
+                select(
+                        concat(k("."),
```

Comment:
Maybe try other delimiters to make sure the result is consistent among implementations. I would suggest at least null, empty, a number and a delimiter coming from another column.

Reply:
Some implementations do not support `NULL` or mixing data types, eg. delimiter is a number and columns are varchars.

Do you think we should throw an error in these cases or just let the DB return the error?